File: js/roman_numerals_helper.js
// https://www.codewars.com/kata/51b66044bce5799a7f000003

class RomanNumerals {
    static toRoman(num) {
      const romanNumerals = [
        { value: 1000, numeral: 'M' },
        { value: 900, numeral: 'CM' },
        { value: 500, numeral: 'D' },
        { value: 400, numeral: 'CD' },
        { value: 100, numeral: 'C' },
        { value: 90, numeral: 'XC' },
        { value: 50, numeral: 'L' },
        { value: 40, numeral: 'XL' },
        { value: 10, numeral: 'X' },
        { value: 9, numeral: 'IX' },
        { value: 5, numeral: 'V' },
        { value: 4, numeral: 'IV' },
        { value: 1, numeral: 'I' }
      ];
  
      let romanNumeral = '';
      for (let i = 0; i < romanNumerals.length; i++) {
        while (num >= romanNumerals[i].value) {
          romanNumeral += romanNumerals[i].numeral;
          num -= romanNumerals[i].value;
        }
      }
  
      return romanNumeral;
    }
  
    static fromRoman(str) {
      const romanNumerals = [
        { numeral: 'M', value: 1000 },
        { numeral: 'CM', value: 900 },
        { numeral: 'D', value: 500 },
        { numeral: 'CD', value: 400 },
        { numeral: 'C', value: 100 },
        { numeral: 'XC', value: 90 },
        { numeral: 'L', value: 50 },
        { numeral: 'XL', value: 40 },
        { numeral: 'X', value: 10 },
        { numeral: 'IX', value: 9 },
        { numeral: 'V', value: 5 },
        { numeral: 'IV', value: 4 },
        { numeral: 'I', value: 1 }
      ];
  
      let num = 0;
      let i = 0;
      while (i < str.length) {
        for (let j = 0; j < romanNumerals.length; j++) {
          if (str.startsWith(romanNumerals[j].numeral, i)) {
            num += romanNumerals[j].value;
            i += romanNumerals[j].numeral.length;
            break;
          }
        }
      }
  
      return num;
    }
  }
  
  console.log(RomanNumerals.fromRoman("XVI"));
  console.log(RomanNumerals.toRoman(1666));

File: python/roman_numerals_helper.py
# https://www.codewars.com/kata/51b66044bce5799a7f000003

class RomanNumerals:
    @staticmethod
    def to_roman(val):
        roman_numerals = {
            1000: 'M',
            900: 'CM',
            500: 'D',
            400: 'CD',
            100: 'C',
            90: 'XC',
            50: 'L',
            40: 'XL',
            10: 'X',
            9: 'IX',
            5: 'V',
            4: 'IV',
            1: 'I'
        }

        roman_numeral = ''
        for num, numeral in roman_numerals.items():
            while val >= num:
                roman_numeral += numeral
                val -= num

        return roman_numeral

    @staticmethod
    def from_roman(roman_num):
        roman_numerals = {
            'M': 1000,
            'CM': 900,
            'D': 500,
            'CD': 400,
            'C': 100,
            'XC': 90,
            'L': 50,
            'XL': 40,
            'X': 10,
            'IX': 9,
            'V': 5,
            'IV': 4,
            'I': 1
        }

        num = 0
        i = 0
        while i < len(roman_num):
            if i + 1 < len(roman_num) and roman_num[i:i + 2] in roman_numerals:
                num += roman_numerals[roman_num[i:i + 2]]
                i += 2
            else:
                num += roman_numerals[roman_num[i]]
                i += 1

        return num

roman = RomanNumerals()

print(roman.to_roman(1666))
print(roman.from_roman('XCVII'))